lib/path.js

```diff
@@ -306,6 +322,11 @@ win32.dirname = function(path) {
 
 
 win32.basename = function(path, ext) {
+  assertPath(path);
+
+  if (ext !== undefined && typeof ext !== 'string')
```

Comment:
This causes breakage when used as an Array.map callback, which will pass in an index for `ext`.

Noticed because I had:
```js
result = directories.map(path.basename);
```

And upgrading now throws:
```
>> TypeError: ext must be a string
>>     at posix.basename (path.js:550:11)
>>     at Array.map (native)
```

Technically an improper use of `basename` but still a small reduction in utility.